staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -454,6 +452,9 @@ func (s *GenericAPIServer) PrepareRun() preparedGenericAPIServer {
 //                             listenerStoppedCh
 //                                     |
 //             HTTPServerStoppedListening (httpServerStoppedListeningCh)
+//                                     |
+//                       (AuditBackend::Shutdown())
```

Comment:
Hmm - that's not what the code is actually doing now - it seems that audit shutdown is happening after drainedCh (InFlightRequestDrained signal):

https://github.com/kubernetes/kubernetes/pull/110026/files#diff-68eb390e148644c38174960be4d103c0e377e71eb540aa8c395f7d2278363973R601